components/login.jsx

```diff
@@ -0,0 +1,112 @@
+var React = require('react');
+var Router = require('react-router');
+var Link = Router.Link;
+
+var teachAPI = require('../lib/teach-api');
+
+var Login = React.createClass({
+  getDefaultProps: function() {
+    return {
+      teachAPI: teachAPI,
+      alert: defaultAlert
+    };
+  },
+  componentDidMount: function() {
+    var teachAPI = this.props.teachAPI;
+
+    teachAPI.on('login:error', this.handleApiLoginError);
+    teachAPI.on('login:cancel', this.handleApiLoginCancel);
+    teachAPI.on('login:success', this.handleApiLoginSuccess);
+    teachAPI.on('logout', this.handleApiLogout);
+    this.setState({username: teachAPI.getUsername()});
+  },
+  componentWillUnmount: function() {
+    var teachAPI = this.props.teachAPI;
+
+    teachAPI.removeListener('login:error', this.handleApiLoginError);
+    teachAPI.removeListener('login:cancel', this.handleApiLoginCancel);
+    teachAPI.removeListener('login:success', this.handleApiLoginSuccess);
+    teachAPI.removeListener('logout', this.handleApiLogout);
```

Comment:
Do we ever need to clear **username**? Sorry I'm reviewing this on my phone right now if you did somewhere within this patch ignore me